compiler/rustc_expand/src/proc_macro_server.rs

```diff
@@ -354,9 +355,8 @@ pub struct Literal {
     span: Span,
 }
 
-pub(crate) struct Rustc<'a> {
-    resolver: &'a dyn ResolverExpand,
-    sess: &'a ParseSess,
+pub(crate) struct Rustc<'a, 'b> {
+    ecx: &'a mut ExtCtxt<'b>,
```

Comment:
```suggestion
    ecx: &'a mut ExtCtxt<'a>,
```
A single lifetime should probably be enough here.

Reply:
I believe I tried this locally and it will not build if we don't use 2 lifetimes. IIRC this is because the inner lifetime is made invariant by the `&mut`, like `&'a mut &'b ()` not being convertible to `&'c mut &'c ()` (https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=2f03be1aec5cd6811df24b957bac7fb5)

Other places which store an `&mut ExtCtxt<'_>` in a struct need to use 2 lifetimes as well (e.g. https://searchfox.org/rust/rev/7c89e389d00cfc7e86ae7e1b45880da4f5f5c9f5/compiler/rustc_expand/src/placeholders.rs#180)